core/lib/multivm/src/versions/vm_m5/transaction_data.rs

```diff
@@ -59,12 +60,22 @@ impl From<Transaction> for TransactionData {
                     U256::zero()
                 };
 
+                // Ethereum transactions do not sign gas per pubdata limit, and so for them we need to use
+                // some default value. We use the maximum possible value that is allowed by the bootloader
+                // (i.e. we can not use u64::MAX, because the bootloader requires gas per pubdata for such
+                // transactions to be higher than MAX_GAS_PER_PUBDATA_BYTE).
+                let gas_per_pubdata_limit = if common_data.transaction_type.is_ethereum_type() {
+                    MAX_GAS_PER_PUBDATA_BYTE.into()
+                } else {
+                    common_data.fee.gas_per_pubdata_limit
+                };
```

Comment:
why do you need to change it for old vms?

Reply:
because in the old VMs `MAX_GAS_PER_PUBDATA_BYTE` was lower than for the new one

Reply:
So basically this `if` is what used to be done implicitly on the API side. But now, since the constant has changed, it has to be done individually for each VM version